test/extended/machines/scale.go

```diff
@@ -0,0 +1,207 @@
+	if err != nil {
+		return fmt.Errorf("error calling getScaleClient: %v", err)
+	}
+
+	scale, err := scaleClient.Scales(machineAPINamespace).Get(schema.GroupResource{Group: machineAPIGroup, Resource: "MachineSet"}, name)
+	if err != nil {
+		return fmt.Errorf("error calling scaleClient.Scales get: %v", err)
+	}
+
+	scaleUpdate := scale.DeepCopy()
+	scaleUpdate.Spec.Replicas = int32(replicas)
+	_, err = scaleClient.Scales(machineAPINamespace).Update(schema.GroupResource{Group: machineAPIGroup, Resource: "MachineSet"}, scaleUpdate)
+	if err != nil {
+		return fmt.Errorf("error calling scaleClient.Scales update while setting replicas to %d: %v", err, replicas)
+	}
+	return nil
+}
+
+var _ = g.Describe("[Feature:Machines][Serial] Managed cluster should", func() {
+	g.It("grow and decrease when scaling different machineSets simultaneously", func() {
+		// expect new nodes to come up for machineSet
+		verifyNodeScalingFunc := func(c *kubernetes.Clientset, dc dynamic.Interface, expectedScaleOut int, machineSet objx.Map) bool {
+			nodes, err := getNodesFromMachineSet(c, dc, machineName(machineSet))
+			if err != nil {
+				e2e.Logf("Error getting nodes from machineSet: %v", err)
+				return false
+			}
+			for i := range nodes {
+				if !isNodeReady(*nodes[i]) {
+					e2e.Logf("Node %q is not ready", nodes[i].Name)
+					return false
+				}
+			}
+			return len(nodes) == expectedScaleOut
+		}
+
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err := dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("checking initial cluster workers size")
+		nodeList, err := c.CoreV1().Nodes().List(metav1.ListOptions{
+			LabelSelector: nodeLabelSelectorWorker,
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		initialNumberOfWorkers := len(nodeList.Items)
+		expectedScaleOut := 3
+
+		g.By("fetching machineSets")
+		machineSets, err := listMachineSets(dc, "")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(machineSets)).To(o.BeNumerically(">", 2))
+		machineSet0 := machineSets[0]
+		initialReplicasMachineSet0 := getMachineSetReplicaNumber(machineSet0)
+		machineSet1 := machineSets[1]
+		initialReplicasMachineSet1 := getMachineSetReplicaNumber(machineSet1)
+
+		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineName(machineSet0), initialReplicasMachineSet0, expectedScaleOut))
+		err = scaleMachineSet(machineName(machineSet0), expectedScaleOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineName(machineSet1), initialReplicasMachineSet1, expectedScaleOut))
+		err = scaleMachineSet(machineName(machineSet1), expectedScaleOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Eventually(func() bool {
+			return verifyNodeScalingFunc(c, dc, expectedScaleOut, machineSet0)
+		}, scalingTime, 5*time.Second).Should(o.BeTrue())
+		o.Eventually(func() bool {
```

Comment:
Can this be simplified to:

```go
o.Eventually(verifyNodeScalingFunc(c, dc, expectedScaleOut, machineSet0), scalingTime, 5*time.Second).Should(o.BeTrue())
```

Reply:
Looking at the implementation, I guess not:

```go
func New(asyncType AsyncAssertionType, actualInput interface{}, fail types.GomegaFailHandler, timeoutInterval time.Duration, pollingInterval time.Duration, offset int) *AsyncAssertion {
	actualType := reflect.TypeOf(actualInput)
	if actualType.Kind() == reflect.Func {
		if actualType.NumIn() != 0 || actualType.NumOut() == 0 {
			panic("Expected a function with no arguments and one or more return values.")
		}
	}
...
```

Reply:
So you'll need to revert the changes you did in https://github.com/openshift/origin/pull/22564/commits/222d477f752080c14de48ce1ecc20b6d0fd7621d, then LGTM.